### PR TITLE
[IMPROVED] Make sure preferred peer for stepdown is healthy.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -1290,21 +1289,42 @@ func (n *raft) StepDown(preferred ...string) error {
 			preferred = nil
 		}
 	}
+	// Can't pick ourselves.
+	if maybeLeader != noLeader && maybeLeader == n.id {
```
Comment (neilalexander) [inline server/raft.go:1293]: Not a big deal but I guess this could just be `if maybeLeader == n.id`

**Review by neilalexander — APPROVED** [review]
LGTM!

**Review by Jarema — APPROVED** [review]
LGTM

File: server/raft.go (hunk 1 above)
Comment (derekcollison) [reply server/raft.go]: Will update. Thanks.

**Review by wallyqs — APPROVED** [review]

**Pull request merged**